frost-core/src/tests/vectors_dkg.rs

```diff
@@ -0,0 +1,226 @@
+
+    let verifying_key =
+        VerifyingKey::deserialize(json_to_element::<C>(&inputs["verifying_key"])).unwrap();
+
+    let signing_share =
+        SigningShare::deserialize(json_to_scalar::<C>(&participant["signing_share"])).unwrap();
+
+    let key_package = KeyPackage {
+        header: Header::default(),
+        identifier: participant_1_id,
+        signing_share,
+        verifying_share,
+        verifying_key,
+        min_signers: 2,
+    };
+
+    DKGTestVectors {
+        secret,
+        coefficient,
+        round1_packages,
+        round2_packages,
+        public_key_package,
+        key_package,
+        participant_id: participant_1_id,
+    }
+}
+
+fn build_round_1_package<C: Ciphersuite>(
+    json_vectors: &Value,
+    participant_num: usize,
+) -> Round1Package<C> {
+    let inputs = &json_vectors["inputs"];
+    let participant = &inputs[participant_num.to_string()];
+    let vss_commitment = participant["vss_commitments"]
+        .as_array()
+        .unwrap()
+        .iter()
+        .map(|v| json_to_element::<C>(v))
+        .collect();
+
+    let commitment = VerifiableSecretSharingCommitment::deserialize(vss_commitment).unwrap();
+
+    let proof_of_knowledge = Signature::deserialize(
+        C::SignatureSerialization::try_from(
+            hex::decode(participant["proof_of_knowledge"].as_str().unwrap()).unwrap(),
+        )
+        .debugless_unwrap(),
+    )
+    .unwrap();
+
+    Round1Package {
+        header: Header::default(),
+        commitment,
+        proof_of_knowledge,
+    }
+}
+
+fn build_round_2_package<C: Ciphersuite>(
+    json_vectors: &Value,
+    sender_num: usize,
+) -> Round2Package<C> {
+    let inputs = &json_vectors["inputs"];
+
+    let signing_share = SigningShare::deserialize(json_to_scalar::<C>(
+        &inputs["1"]["signing_shares"][sender_num.to_string()],
+    ))
+    .unwrap();
+
+    Round2Package {
+        header: Header::default(),
+        signing_share,
+    }
+}
+
+fn build_public_key_package<C: Ciphersuite>(json_vectors: &Value) -> PublicKeyPackage<C> {
+    let inputs = &json_vectors["inputs"];
+
+    let mut verifying_shares = BTreeMap::new();
+
+    let max_participants = json_vectors["config"]["MAX_PARTICIPANTS"].as_u64().unwrap() as u8;
+
+    for i in 1..=max_participants {
+        let participant_id: Identifier<C> = (inputs[i.to_string()]["identifier"].as_u64().unwrap()
+            as u16)
+            .try_into()
+            .unwrap();
+        let verifying_share = VerifyingShare::deserialize(json_to_element::<C>(
+            &inputs[i.to_string()]["verifying_share"],
+        ))
+        .unwrap();
+        verifying_shares.insert(participant_id, verifying_share);
+    }
+
+    let verifying_key =
+        VerifyingKey::deserialize(json_to_element::<C>(&inputs["verifying_key"])).unwrap();
+
+    PublicKeyPackage {
+        header: Header::default(),
+        verifying_shares,
+        verifying_key,
+    }
+}
+
+/// TODO documentation for test
```

Comment:
Do you still want to document this?

Reply:
Done in [b049dfa](https://github.com/ZcashFoundation/frost/pull/587/commits/b049dfa139cb74a73de927e5ca8d24a662639d38)